src/core/packages/saved-objects/api-server-internal/src/lib/utils/decorate_es_error.ts

```diff
@@ -38,7 +38,10 @@ export function decorateEsError(error: EsErrors) {
     throw new Error('Expected an instance of Error');
   }
 
-  const { reason } = get(error, 'body.error', { reason: undefined }) as { reason?: string };
+  const { reason: esErrorReason } = get(error, 'body.error', { reason: undefined }) as {
+    reason?: string | null;
+  };
```

Comment:
nit: this could have been caught in the past if we used a type provided by the ES client. 

Maybe `ErrorCause` instead of the explicit casting to `{ reason?: string | null }`?